### DISCUSSION TITLE
Command line tool rather in addition/rather than template?

### DISCUSSION THREAD

**Discussion by davidwinter** [cawa-93/vite-electron-builder#275] [Ideas]
I was wondering if it has been considered to create a command line tool that has the relevant configs, scripts etc, and defines a default directory structure for source files to be found? This would mean if changes to those scripts or configs were made, the command line tool could be upgraded, without the need to manually diff/update files used in this template? It was one of the benefits of `electron-webpack`.

**Comment by cawa-93** [comment]
Thanks for the question. No, I do not plan to create command line tool. I position this template only as a starting point. After starting your project, there is no need to follow the changes in the template to transfer diff/updates to your project.

In addition, in my opinion, the configuration of the Vite is quite simple and a separate tool for this may be superfluous.

**Comment by StarLederer** [comment]
I was making a [Svelte version](https://github.com/HermanLederer/vite-electron-builder-svelte) of this template today and though about making a CLI tool to just initiate this with the framework of choice and maybe optional scss support, linters and tests. I really like the structure of the template and think it is pretty easy to do things like this manually but I also remember myself half a year ago being barely able to set up electron without any devtools bundlers and especially frameworks.

Perhaps an npm initializer could be useful to make it easier for newcomers?

I'm thinking something like

```bash
npm init vite-electron-builder my-vite-electron-app --template vue --scss --test playwright
```

I have never made npm initilizers before but I would be happy to try to make it, if it gains support from the author.

**Reply by cawa-93** [reply]
Here are nice example
https://github.com/vitejs/vite/tree/main/packages/create-vite
